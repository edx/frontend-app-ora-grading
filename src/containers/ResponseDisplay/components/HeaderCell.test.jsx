import React from 'react';
import { shallow } from 'enzyme';

import {
  HeaderEllipsesCell,
  HeaderFileExtensionCell,
  HeaderInfoPopoverCell,
} from './HeaderCell';

jest.mock('components/InfoPopover', () => 'InfoPopover');

describe('HeaderEllipsesCell', () => {
  describe('component', () => {
    const props = {
      value: 'some test text value',
    };
    let el;
    beforeEach(() => {
      el = shallow(<HeaderEllipsesCell {...props} />);
    });
    test('snapshot', () => {
      expect(el).toMatchSnapshot();
    });

    describe('behavior', () => {
      test('content', () => {
        expect(el.text()).toEqual(props.value);
      });
    });
  });
});

describe('HeaderFileExtensionCell', () => {
  describe('component', () => {
    const props = {
      value: 'file_name.with_extension.pdf',
    };
    let el;
    beforeEach(() => {
      el = shallow(<HeaderFileExtensionCell {...props} />);
    });
    test('snapshot', () => {
      expect(el).toMatchSnapshot();
    });

    describe('behavior', () => {
      test('content', () => {
        expect(el.text()).toEqual('PDF');
      });
    });
  });
});

describe('HeaderInfoPopoverCell', () => {
  describe('component', () => {
    const props = {
      row: {
        original: {
          key: 'value',
          key2: 'value2',
        },
      },
    };
    let el;
    beforeEach(() => {
      el = shallow(<HeaderInfoPopoverCell {...props} />);
    });
    test('snapshot', () => {
      expect(el).toMatchSnapshot();
    });

    describe('behavior', () => {
      test('content', () => {
        const { original } = props.row;
        Object.keys(original).forEach(key => {
          expect(el.text()).toContain(key.toUpperCase());
          expect(el.text()).toContain(original[key]);
        });
      });
    });
  });
});
