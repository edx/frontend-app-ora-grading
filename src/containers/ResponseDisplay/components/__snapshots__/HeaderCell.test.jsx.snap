// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderEllipsesCell component snapshot 1`] = `
<div
  className="text-truncate"
>
  some test text value
</div>
`;

exports[`HeaderFileExtensionCell component snapshot 1`] = `
<div
  className="text-truncate"
>
  PDF
</div>
`;

exports[`HeaderInfoPopoverCell component snapshot 1`] = `
<InfoPopover>
  <div
    className="help-popover-option"
    key="key"
  >
    <strong>
      KEY
    </strong>
    <br />
    value
  </div>
  <div
    className="help-popover-option"
    key="key2"
  >
    <strong>
      KEY2
    </strong>
    <br />
    value2
  </div>
</InfoPopover>
`;
